### PR TITLE
Feature #24437 variants

### PR DESCRIPTION
**Pull request by WHLukasz**
created new documentation page for image variants and image transformations

image

added description for transforming images, both via creating variants and transforming images while fetching them. Added examples for newly added endpoints

Note that the documentation sections for designated endpoints for creating and removing variants may be removed in the future, if the corresponding endpoints get removed

### CODE REVIEW

File: docs/API/variants.md [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+            "height":200
+        }
+    }
+    ```
+    { data-search-exclude }
+
+!!! Note
+    - Media objects without any variants initially won't have the "variants" key in their object data. It will be automatically added when the first variant for that media is created.
+    - Removing all variants of an asset won't delete the "variants" key from object data.
+    - Variant names must be unique and follow the regex pattern: `^[_a-z]+$` (case insensitive).
```
Comment (CiotkaCierpienia) [inline docs/API/variants.md:59]:
```suggestion
    - Variant names must be unique and follow the regex pattern: `^[_a-zA-Z0-9]+$` (case insensitive).
```
This was changed as we do not need to restrict the name as much.

File: docs/API/variants.md [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+
+## Transformation Types
+
+Currently, the only acceptable type of transformation is `trim`.
+<!-- The transformations follow Cloudflare's transformation standards. -->
+<!-- https://developers.cloudflare.com/images/transform-images/transform-via-workers -->
```
Comment (CiotkaCierpienia) [inline docs/API/variants.md:65]: Please remove that

File: docs/API/variants.md [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+    { data-search-exclude }
+
+    If the original image is 400/400, it will result in cropping a 50/50 square in the top left corner, 10 pixels away from the top/left side.
+
+!!! Note
+    Transforming works together with resizing (width/height properties used in every GET image endpoint). Note that images are transformed before they are resized.
+
+## Handling Variants
+
+Creating, removing and updating variants can be done in two ways, either by updating the media content object itself using standard endpoints for editing the content object's data or by using designated endpoints for operations on variants.
```
Comment (CiotkaCierpienia) [inline docs/API/variants.md:114]: Please remove mentions of designated endpoints, as they are not needed and will be removed for sure

**Review by CiotkaCierpienia — COMMENTED** [review]
Please also add information about variants in the dashboard section of the documentation as described in the ticket

**Comment by WHLukasz** [comment]
added `variants` section to deashboard/media-library

image

**Comment by CiotkaCierpienia** [comment]
Everything is ok, merge is waiting for backend build

**Pull request merged**